src/main/java/io/ipfs/api/IPFS.java

```diff
@@ -490,6 +492,28 @@ public Object log() throws IOException {
         }
     }
 
+    private boolean versionIsValid(String version) {
+        String[] parts = version.split("\\.");
```

Comment:
@ianopolous is there a reason you cross-compile this instead of using ipfs/js-ipfs-api?